pymc/tests/test_model.py

```diff
@@ -702,6 +703,61 @@ def test_nested_model_coords():
     assert set(m2.RV_dims) < set(m1.RV_dims)
 
 
+def test_shape_error_from_fixed_dims():
+    """
+    with #5751 dimensions set by add_coord() are default TensorConstant type
+    and so have fixed dimensions. Trying to resize them will raise a ShapeError,
+    which this test checks for.
+    """
+    with pm.Model() as pmodel:
+        a = pm.Normal("a", mu=[1, 2, 3], dims="fixed")
+        m = pm.MutableData("m", [[1, 2, 3]], dims=("one", "fixed"))
+
+    with pytest.raises(ShapeError):
```

Comment:
Included from the test below, to reduce test runtime.
```suggestion
        m = pm.MutableData("m", [[1, 2, 3]], dims=("one", "fixed"))
        
        # This is fine because the "fixed" dim is not resized
        pm.set_data({"m": [[1, 2, 3], [3, 4, 5]]})

    with pytest.raises(ShapeError):
```